File: ruby/format_duration.rb
def format_duration(seconds)
  s = 's'
  duration = {
    year: (seconds / 24 / 60 / 60 / 365).floor,
    day: (seconds / 24 / 60 / 60 % 365).floor,
    hour: (seconds / 60 / 60 % 24).floor,
    minute: (seconds / 60 % 60).floor,
    second: (seconds % 60)
  }

  english = duration
            .map { |time, value| !value.zero? ? "#{value} #{time}#{value > 1 ? s : ''}" : '' }
            .reject { |time| time == '' }
  english.length > 1 ? "#{english[0..-2].join(', ')} and #{english[-1]}" : english[0] || 'now'
end

File: js/formatDuration.js
const formatDuration = (seconds) => {
  const duration = {
    year: Math.floor(seconds / (24 * 3600 * 365)),
    day: Math.floor((seconds % (24 * 3600 * 365)) / (24 * 3600)),
    hour: Math.floor((seconds % (24 * 3600)) / 3600),
    minute: Math.floor((seconds % 3600) / 60),
    second: seconds % 60,
  };
  const english = Object.keys(duration)
    .map((t) => (duration[t] ? `${duration[t]} ${t}` : '') + (duration[t] > 1 ? 's' : ''))
    .filter((e) => e);

  return english.length > 1
    ? `${english.slice(0, -1).join(', ')} and ${english.slice(-1)}`
    : english.join('')
    || 'now';
};
